### ISSUE TITLE
Support diff highlight with original file format syntax scheme

### ISSUE THREAD

**Issue by asfaltboy** [divmain/GitSavvy#961]
This might be dupe, if so I appologize; I saw the discussion being had a number of times, but never as a proper feature request, so here we go.

In Github the diff with highlight [is displayed very nicely][1], and I'd like to have it in sublime.

I know this can be done, but not sure how cleanly the code will remain after. [The plugin sublime-compare-side-by-side][2] does this by [applying "highlight"][3] to files that have been added to the view with the original/default syntax scheme. I understand that with side-by-side diff it is easier to keep syntax, and in "patch view" case we might need to apply the syntax line by line (by checking the original file's syntax for the row) + apply the diff highlight.

Related tickets, discussions:
https://github.com/divmain/GitSavvy/issues/666
https://github.com/divmain/GitSavvy/pull/661/commits/8c1313be155ba73c8661aeed1fce2d1e5e6fae84#r121255619

[1]: https://github.com/divmain/GitSavvy/pull/960/commits/07d701fb029cda7742b2cdb3ad6a40bf86146679
[2]: https://github.com/DougTy/sublime-compare-side-by-side
[3]: https://github.com/DougTy/sublime-compare-side-by-side/blob/master/sbs_compare.py#L242-L269

**Comment by randy3k** [comment]
Actually, I tried the approach of applying the syntax line by line in the blame view, but it was incredibly slow and I had to abandon that.

**Comment by stoivo** [comment]
Have you tried to do it only for the content vissable? I think sublimes has an API for checking the viewport. Or is it slow on small files as well

**Comment by randy3k** [comment]
It was ok for small files.

**Comment by stoivo** [comment]
And did you look at the option for only doing it for the current virwport?

**Comment by randy3k** [comment]
No. I am not sure if sublime has an event listener on change of viewport.

**Comment by asfaltboy** [comment]
We might be able to get the syntax for the whole file then, by using the "open, get syntax and close quickly" method. To avoid continuous flicker, we can also "cache" the syntax for each opened file, as described here: https://forum.sublimetext.com/t/get-syntax-associated-with-file-extension/26348/11

**Comment by asfaltboy** [comment]
There's also a discussion on a relevant feature request in SublimeTextIssues/Core#2312, and [a workaround provided  by sublime_lib here][1].

[1]: https://github.com/SublimeText/sublime_lib/blob/master/st3/sublime_lib/syntax.py#L104

**Comment by kaste** [comment]
@randy3k How did you apply the syntax line-by-line? Applying foreground regions token by token? Which tokenizer did you use?

**Comment by randy3k** [comment]
I used `view.scope_name` to extract the scopes from an ordinary view of the file and then use `view.add_regions` to apply the scopes to the blame view. By it was quite slow for files > 1000 lines.

**Comment by asfaltboy** [comment]
Do you still have that branch? Maybe we can hack around it?

**Comment by randy3k** [comment]
I have discarded that unfortunately.
